sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,237 @@
+        elif self.transformer is not None:
+            self.transformer_ = clone(self.transformer)
+        else:
+            self.transformer_ = FunctionTransformer(
+                func=self.func, inverse_func=self.inverse_func, validate=False)
+        # XXX: sample_weight is not currently passed to the
+        # transformer. However, if transformer starts using sample_weight, the
+        # code should be modified accordingly. At the time to consider the
+        # sample_prop feature, it is also a good use case to be considered.
+        self.transformer_.fit(y)
+        if self.check_inverse:
+            random_state = check_random_state(self.random_state)
+            n_subsample = min(10, y.shape[0])
+            subsample_idx = random_state.choice(range(y.shape[0]),
+                                                size=n_subsample,
+                                                replace=False)
+            if not np.allclose(
+                    y[subsample_idx],
+                    self.transformer_.inverse_transform(
+                        self.transformer_.transform(y[subsample_idx]))):
+                raise ValueError("The provided functions or transformer are"
+                                 " not strictly inverse of each other. If"
+                                 " you are sure you want to proceed regardless"
+                                 ", set 'check_inverse=False'")
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,) optional
+            Array of weights that are assigned to individual samples.
+            If not provided, then each sample is given unit weight.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(X, y, multi_output=True, y_numeric=True)
+
+        # transformer are designed to modify X which is a 2d dimensional, we
+        # need to modify y accordingly.
+        if y.ndim == 1 and self.func is None:
+            y_2d = y.reshape(-1, 1)
+        else:
+            y_2d = y
+        self._fit_transformer(y_2d)
+
+        if self.regressor is None:
+            from ..linear_model import LinearRegression
+            self.regressor_ = LinearRegression()
+        else:
+            self.regressor_ = clone(self.regressor)
+
+        support_sample_weight = has_fit_parameter(self.regressor_,
+                                                  'sample_weight')
+        if not support_sample_weight and sample_weight is not None:
+            raise ValueError("The regressor {} does not support sample "
```

Comment:
Bagging is a bit special because it changes its algorithm if the base estimator supports weight, even when they were not provided by the user.

For a paradigm where you just pass the parameter if you have it, see CalibratedClassifierCV, SelectFromModel. But I see what you mean that this is commonly used... I'm just pretty sure it'll break with sample props.